x/staking/keeper/slash.go

```diff
@@ -129,8 +128,16 @@ func (k Keeper) Slash(ctx sdk.Context, consAddr sdk.ConsAddress, infractionHeigh
 
 	// Deduct from validator's bonded tokens and update the validator.
 	// Burn the slashed tokens from the pool account and decrease the total supply.
+	initialLiquidTokens := validator.TokensFromShares(validator.LiquidShares).TruncateInt()
 	validator = k.RemoveValidatorTokens(ctx, validator, tokensToBurn)
 
+	// Proportionally deduct any liquid tokens from the global total
+	updatedLiquidTokens := validator.TokensFromShares(validator.LiquidShares).TruncateInt()
+	slashedLiquidTokens := initialLiquidTokens.Sub(updatedLiquidTokens)
+	if err := k.DecreaseTotalLiquidStakedTokens(ctx, slashedLiquidTokens); err != nil {
+		panic(err)
```

Comment:
Add docstring explaining why it's okay to panic here. IIUC, this should never happen as long as the state is not corrupted, right?

Reply:
that's right - it would indicate that the validator had liquid tokens that were not accounted for in the global total.
Added better clarification [[here](https://github.com/cosmos/cosmos-sdk/pull/16747/commits/53cca311be752f3851e1fa0bbb213dac5dfba6cf)]